web/client/components/geostory/builder/SectionsPreview.jsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2019, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React from 'react';
+import { capitalize } from "lodash";
+import ContainerDimensions from 'react-container-dimensions';
+import { Glyphicon } from 'react-bootstrap';
+import Toolbar from '../../misc/toolbar/Toolbar';
+
+import SideGrid from '../../misc/cardgrids/SideGrid';
+import SideCard from '../../misc/cardgrids/SideCard';
+
+/**
+ * Renders icon for the Section or content provided
+ * @param {object} content content or section of the type of the icon you want to render
+ */
+const Icon = ({ type } = {}) => {
+    const glyphs = {
+        text: 'sheet'
+    };
+    return <Glyphicon glyph={glyphs[type] || 'question-sign'} />;
+};
+
+/**
+ * Preview view for a geostory-content
+ * @param {object} [content={}] content that have to render preview
+ */
+const Preview = ({ width } = {}) => {
+    const res = 1080 / 1920;
+    const height = width * res;
+    return (
+        <div style={{ position: 'relative', width, height }}>
+            <div style={{ position: 'absolute', width: '100%', height: '100%', backgroundColor: '#dddddd', display: 'flex' }}>
+                <Glyphicon glyph="1-map" style={{ fontSize: height / 2, margin: 'auto' }} />
+            </div>
+        </div>
+    );
+};
+
+/**
+ * Transforms a geostory section into a SideGrid item.
+ * @param {object} section the section to transform
+ */
+const sectionToItem = ({
+    contents,
+    type,
+    title,
+    id
+}) => {
+    return {
+        id: id,
+        preview: <Icon type={type} />,
+        tools: <Toolbar
+            btnDefaultProps={{
+                className: 'square-button-md no-border'
+            }}
+            buttons={[
+                {
+                    glyph: 'zoom-to',
+                    visible: contents.length === 1,
+                    tooltip: 'Zoom to content'
+                }
+            ]} />,
+        title: title,
+        description: `type: ${type}`,
+
+        body: contents
+            ? <div style={{ position: 'relative' }}>
```

Comment:
I think this position relative was needed for ContainerDimansion, it had strange behaviours, I don't know it it really needed now.